src/test/CharterManager-unit.t.sol

```diff
@@ -386,6 +378,60 @@ contract CharterManagerTest is TestBase {
         assertEq(a.gems(), 100 * 1e18);
     }
 
+    function test_exit() public {
+        (Usr a,) = init_user();
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+
+        a.join(200 * 1e6);
+        assertEq(a.gems(), 200 * 1e18);
+        assertEq(gem.balanceOf(address(a)), 0);
+
+        // check exit of unlocked gems does not affect the vault and does not prevent increasing debt
+        a.exit(200 * 1e6);
+        assertEq(a.gems(), 0);
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+    }
+
+    function test_exit_other() public {
+        (Usr a, Usr b) = init_user();
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+
+        a.join(200 * 1e6);
+        assertEq(a.gems(), 200 * 1e18);
+        assertEq(gem.balanceOf(address(a)), 0);
+
+        a.hope(address(b));
+        b.flux(address(a), address(b), 200 * 1e18);
+        b.exit(address(a), 200 * 1e6);
+        assertEq(a.gems(), 0);
+        assertEq(b.gems(), 0);
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+    }
+
+    function testFail_exit_other() public {
+        (Usr a, Usr b) = init_user();
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+
+        a.join(200 * 1e6);
+        assertEq(a.gems(), 200 * 1e18);
+        assertEq(gem.balanceOf(address(a)), 0);
+
+        // b can not exit a's funds
+        b.exit(address(a), 200 * 1e6);
+    }
+
+    function testFail_flux_before_exit_other() public {
```

Comment:
```suggestion
    function testFail_flux_from_other() public {
```

Reply:
I think this should just be a test for showing when `flux` fails doing from other. `exit` doesn't really have much to do with this test.

Reply:
Ok, also here the intention was to show that when there is no hoping, the flux and exit sequence fails.
But since we removed the original success test I'm removing this as well.

There are alread `test_flux_to_other`, and `testFail_flux_from_other1` that show the flux from other fails without hoping.